core/include/bitarithm.h

```diff
@@ -172,6 +172,10 @@ static inline unsigned bitarithm_lsb(unsigned v)
 {
 /* Source: http://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup */
     extern const uint8_t MultiplyDeBruijnBitPosition[32];
+    /* cppcheck-suppress oppositeExpression
+     * (reason: `x & -x` is a bit twiddling idiom to extract the LSB; the
+     * check assumes that all arguments to binary operations should have the
+     * same sign) */
```

Comment:
Actually, this is not about the same sign. I changed the wording a bit, please confirm.

Reply:
Yeah, this is better, the old wording was sloppy.